src/org/ods/services/AquaService.groovy

```diff
@@ -0,0 +1,64 @@
+package org.ods.services
+
+import org.ods.util.ILogger
+
+class AquaService {
+
+    private final def script
+    private final ILogger logger
+
+    AquaService(def script, ILogger logger) {
+        this.script = script
+        this.logger = logger
+    }
+
+    void scanViaCli(String aquaUrl, String registry, String imageRef, String credentialsId, String reportFile) {
+        logger.info "Starting to scan via Aqua CLI..."
+        int status = 0
+        withCredentials(credentialsId) { username, password ->
+            status = script.sh(
+                label: 'Scan via Aqua CLI',
+                returnStatus: true,
+                script: """
+                  scannercli scan ${imageRef} \
+                  --dockerless \
+                  --register \
+                  --text \
+                  --htmlfile '${reportFile}' \
+                  -U '${username}' \
+                  -P '${password}' \
+                  -H '${aquaUrl}' \
+                  --registry '${registry}'
+                """
+            )
+        }
+        // see possible status codes at https://docs.aquasec.com/docs/scanner-cmd-scan#section-return-codes
+        switch (status) {
+            case 0:
+                logger.info "Finished scan via Aqua CLI successfully!"
+                break
+            case 1:
+                script.error "An error occurred in processing the scan request " +
+                    "(e.g. invalid command line options, image not pulled, operational error)."
+                break
+            case 4:
+                script.error "The image scanned failed at least one of the Image Assurance Policies specified."
+                break
+            default:
+                script.error "An unknown status code was returned!"
+        }
```

Comment:
I'm wondering if we shouldn't move the responsibility to stop the pipeline to the stage? That would make the service easier to use in case you want to handle failure differently.

Reply:
moved it